drivers/counter/counter_nrfx_rtc.c

```diff
@@ -81,18 +94,97 @@ static u32_t counter_nrfx_read(struct device *dev)
 	return nrfx_rtc_counter_get(&get_nrfx_config(dev)->rtc);
 }
 
+/* Function calculates distance between to values assuming that one first
+ * argument is in front and that values wrap.
+ */
+static u32_t tick_diff(u32_t val, u32_t old, u32_t top)
+{
+	if (top & (top + 1)) {
+		/* if top is not 2^n-1 */
+		u32_t diff;
+
+		diff = (val > old) ? (val - old) : val + top + 1 - old;
+
+		return diff;
+	}
+
+	return (val - old) & top;
+}
+
+static int set_cc(struct device *dev, u8_t chan, u32_t val)
+{
+	const struct counter_nrfx_config *nrfx_config = get_nrfx_config(dev);
+	const nrfx_rtc_t *rtc = &nrfx_config->rtc;
+	NRF_RTC_Type  *reg = rtc->p_reg;
+	nrf_rtc_int_t int_mask;
+	nrf_rtc_event_t evt;
+	u32_t prev_val;
+	u32_t top;
+	u32_t now;
+
+	evt = (nrf_rtc_event_t)(offsetof(NRF_RTC_Type, EVENTS_COMPARE[chan]));
+	int_mask = (nrf_rtc_int_t)(NRF_RTC_INT_COMPARE0_MASK << chan);
+
+	nrf_rtc_event_clear(reg, evt);
+	top = counter_get_top_value(dev);
+	prev_val = nrf_rtc_cc_get(reg, chan);
+	nrf_rtc_cc_set(reg, chan, val);
+
+	/* If prev CC setting equals or is just in front of the counter then
+	 * there is a risk that before new CC will be set after enabling event
+	 * previous CC will generate COMPARE event. In such risk delay must be
+	 * introduced between writing CC value and enabling the event.
+	 */
+	now = nrf_rtc_counter_get(reg);
+	if (tick_diff(prev_val, now, top) == 1) {
+		LOG_INST_INF(nrfx_config->log,
+			"Delay introduced due to risk of pre-firing.");
+		NRFX_DELAY_US(LF_TICK_US);
+	}
+
+	nrf_rtc_event_enable(reg, int_mask);
+
+	/* If value is set to now + 1 wait one LFCLK cycle to ensure that it is
+	 * written to RTC register.
```

Comment:
Sorry, I don't understand the need of this waiting. Why does it matter if the value is written to the RTC register (I assume you mean the one in LFCLK domain) or not at this point?